proto/evnode/v1/evnode.proto

```diff
@@ -93,6 +93,7 @@ message Metadata {
 message Data {
   Metadata       metadata = 1;
   repeated bytes txs      = 2;
+  bytes          cached_hash = 3; // Optional cached hash for performance
```

Comment:
Do we really need to add this? Just caching the last one in the manager is sufficient right? Sure it is less optimized that cache it for all, but the logic that omits the cache for the serialization can be confusing if you use the api directly .

Reply:
don't we only care about height-1 anyway?